pkg/cloud/services/networking/floatingip.go

```diff
@@ -30,28 +30,28 @@ import (
 func (s *Service) GetOrCreateFloatingIP(openStackCluster *infrav1.OpenStackCluster, ip string) (*floatingips.FloatingIP, error) {
 	var fp *floatingips.FloatingIP
 	var err error
+	fpCreateOpts := &floatingips.CreateOpts{
+		FloatingNetworkID: openStackCluster.Status.ExternalNetwork.ID,
+	}
+
 	if ip != "" {
 		fp, err = checkIfFloatingIPExists(s.client, ip)
 		if err != nil {
 			return nil, err
 		}
-	}
-	if fp == nil {
-		fpCreateOpts := &floatingips.CreateOpts{
-			FloatingNetworkID: openStackCluster.Status.ExternalNetwork.ID,
-		}
-		if ip != "" {
-			// only admin can add ip address
-			fpCreateOpts.FloatingIP = ip
-		}
-		fp, err = floatingips.Create(s.client, fpCreateOpts).Extract()
-		if err != nil {
-			record.Warnf(openStackCluster, "FailedCreateFloatingIP", "Failed to create floating IP %s: %v", fp.FloatingIP, err)
-			return nil, err
+		if fp != nil {
+			return fp, nil
 		}
-		record.Eventf(openStackCluster, "SuccessfulCreateFloatingIP", "Created floating IP %s with id %s", fp.FloatingIP, fp.ID)
+		// only admin can add ip address
+		fpCreateOpts.FloatingIP = ip
+	}
 
+	fp, err = floatingips.Create(s.client, fpCreateOpts).Extract()
+	if err != nil {
+		record.Warnf(openStackCluster, "FailedCreateFloatingIP", "Failed to create floating IP %s: %v", ip, err)
+		return nil, err
 	}
```

Comment:
```suggestion
	}

```

Reply:
Done. Thank you for the very fast feedback :)